src/chainrules.jl

```diff
@@ -0,0 +1,19 @@
+function rrule(::typeof(*), A::LinearMap, x::AbstractVecOrMat)
+    y = A*x
+    function pullback(dy)
+      DY = unthunk(dy)
+      # Because A is an abstract map, the product is only differentiable w.r.t the input
+      return NoTangent(), NoTangent(), @thunk(A' * DY)
+    end
+    return y, pullback
+end
+
+function rrule(A::LinearMap, x::AbstractVecOrMat)
+    y = A(x)
```

Comment:
```suggestion
    y = A*x
```
Perhaps it will work as is, but I recommend we keep reminding everyone that this is a linear map by using multiplication rather than a function call...